### PR TITLE
Final Project review request.

### PR DESCRIPTION
**Pull request by daudi13**
Hello reviewer,
This projet was executed by @daudi13 & @Abenezer-Tilahun. Below are the project requirements we implemented.

Project requirements

- You should follow the layout of the wireframes provided. You should personalize the rest of the design including colors, typographies, spacings, etc.
- Login page and registration page:
   - Should be built with Devise.
- Food list:
   - Should display a list of food added by the logged-in user as in the wireframe (for teams with only 2 people - display also quantity of a given food).
   - Should lead to a form that allows users to add new food.
- Recipes list:
   - Should display a list of recipes created by the logged-in user as in the wireframe.
   - Should lead to recipe details.
   - If the user is the owner of the recipe, should allow the user to delete it.
- Public recipe list:
   - Should display a list of all public recipes ordered by newest as in the wireframe.
   - Should lead to recipe details.
- Recipe details:
   - If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
   - If the user is the owner of the recipe, should lead to the form that allows the user to add new food.
- General shopping list view (only for a group that has less than 3 members):
   - Should show the list of food that is missing for all recipes of the logged-in user (compare the list of food for all  recipes with the general food list of that user).
   - Should count the total food items and total price of the missing food.
- Make sure there are no N+1 queries happening.
- Create a navigation menu that allows users to open all of the pages you created.

---
Thank you for taking the time to review our project

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,69 @@
...
+> The Recipe app keeps track of all your recipes, ingredients, and inventory. It will allow you to save ingredients, keep track of what you have, create recipes, and generate a shopping list based on what you have and what you are missing from a recipe. Also, since sharing recipes is an important part of cooking the app  allowS you to make them public so anyone can access them.
 
-Things you may want to cover:
+![](./Screenshot%20from%202022-07-02%2007-51-17.png)
 
-* Ruby version
+## EDR diagram
 
-* System dependencies
+![](./recipe_erd_2_members.png)
```
Comment (MahmoudBakr23) [inline README.md:9]: - [ ] Please attach a link to the recorded presentation video of the project. 🙏

File: README.md [hunk 2]
```diff
@@ -1,24 +1,69 @@
...
```
Comment (MahmoudBakr23) [inline README.md:10]: - [ ] Please make sure to deploy your app and attach the live link in the README file besides the presentation video. 😄

File: README.md [hunk 3]
```diff
@@ -1,24 +1,69 @@
...
 
-* Configuration
+## Getting started
 
-* Database creation
+- Run `bundle install` to get the dependecies
+- Run `rails s` to start  the server
```
Comment (MahmoudBakr23) [inline README.md:14]: - [ ] Please add the first two CLIs `rails db:create` and `rails db:migrate` as part of the steps to install and run the app locally. 🙏

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello team 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Well done setting up the controllers
- Great work adding the associations and the validations to your models
- Well done setting up the routes
- Great test suits
- Descriptive PR and commit messages
- Well done deploying your app
- Nice design

However, there is always room for improvement.

> Keep in mind that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

N/A

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 3 above)
Comment (Abenezer-Tilahun) [reply README.md:16]: Done

File: README.md (hunk 2 above)
Comment (daudi13) [reply README.md:10]: @MahmoudBakr23  Thanks for your feedback but I've gone through the project requirements and I haven't been able to see the above requirement. Please let me know if I missed something. Thank you.

File: README.md (hunk 1 above)
Comment (daudi13) [reply README.md:9]: @MahmoudBakr23  Thank you for your feedback but this requirement was not specific to this project. Let me know if we missed something. Thanks

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @daudi13, and @Abenezer-Tilahun👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️
- Good commit history. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**